File: solutions/ruby/atbash-cipher/atbash_cipher.rb
module Atbash
  LETTERS = ('a'..'z').to_a.freeze
  DIGITS = ('0'..'9').to_a.freeze
  ALNUM = LETTERS + DIGITS

  def self.convert(data)
    data.downcase.tr(LETTERS.join, LETTERS.reverse.join).chars.select { |i| ALNUM.include?(i) }
  end

  def self.decode(data)
    convert(data).join
  end

  def self.encode(data)
    convert(data).each_slice(5).map(&:join).join(' ')
  end
end

File: solutions/python/atbash-cipher/atbash_cipher.py
import string

mapping = dict(zip(string.ascii_lowercase, string.ascii_lowercase[::-1]))
mapping.update({i: i for i in string.digits})


def encode(text):
    text = decode(text)
    return " ".join("".join(text[i:i + 5]) for i in range(0, len(text), 5))


def decode(text):
    text = text.lower()
    text = (t for t in text if t.isalnum())
    text = [mapping[t] for t in text]
    return "".join(text)
